src/vs/workbench/contrib/extensions/browser/fileBasedRecommendations.ts

```diff
@@ -153,7 +155,15 @@ export class FileBasedRecommendations extends ExtensionRecommendations {
 		const matchedRecommendations: IStringDictionary<IFileOpenCondition[]> = {};
 		const unmatchedRecommendations: IStringDictionary<IFileOpenCondition[]> = {};
 		let listenOnLanguageChange = false;
-		const languageId = model.getLanguageId();
+		let languageId = model.getLanguageId();
+
+		// Avoid language-specific recommendations for untitled files when language is auto-detected.
+		if (uri.scheme === Schemas.untitled) {
+			const untitledModel = this.untitledTextEditorService.get(uri);
+			if (untitledModel && !untitledModel.hasLanguageSetExplicitly) {
+				languageId = PLAINTEXT_LANGUAGE_ID;
+			}
+		}
```

Comment:
I'm not sure about this.

1. I'm not sure setting the languageId to plaintext is the right approach, I think instead we should update the logic to say that the language just doesn't match, which I believe is the `isLanguageCondition` part in the for loop
2. I wonder if we could still keep some of this auto-detection around... like if the file is more than... 1000 characters or something, chances are the language detection is pretty good.